### PR TITLE
DOC: add FAQ section on zero gradients for rank-based operations

### PR DESCRIPTION
**Pull request by jakevdp**
Added this because similar questions have come up a few times in recent months (most recently in https://github.com/google/jax/discussions/11033)

Preview the new docs here: https://jax--11040.org.readthedocs.build/en/11040/faq.html#gradients-are-zero-for-rank-based-functions

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -693,6 +693,85 @@ Additional reading:
   * `How to avoid NaN gradients when using where <https://github.com/tensorflow/probability/blob/master/discussion/where-nan.pdf>`_.
 
 
+Gradients are zero for rank-based functions
```
Comment (skye) [inline docs/faq.rst:696]: I have to admit it bothers me that our FAQ doesn't contain actual questions :P But maybe that's for a different PR to resolve.

File: docs/faq.rst [hunk 2]
```diff
@@ -693,6 +693,85 @@ Additional reading:
...
+-------------------------------------------
+
+If you define a function that processes the input using rank-based operators
```
Comment (skye) [inline docs/faq.rst:699]: I don't know what "rank-based" means :(

**Review by skye — APPROVED** [review]
This is great!! Very good explanation, even when I don't know what a rank-based function is

File: docs/faq.rst (hunk 2 above)
Comment (jakevdp) [reply docs/faq.rst:699]: I tried to give examples later in the sentence – anything I could do to make that more clear?

Comment (jakevdp) [reply docs/faq.rst:699]: To elaborate, what I had in mind is functions like `greater`, `max`, `argsort`, etc. whose output depends on comparing the order/ranking of inputs.

Comment (skye) [reply docs/faq.rst:699]: Ooh I was thinking rank like the number of dimensions. I think "sort-order-based functions" would be clearer to me, and/or put "whose output depends on comparing the sort-order/ranking of inputs" around here.

File: docs/faq.rst (hunk 1 above)
Comment (jakevdp) [reply docs/faq.rst]: Fixed, at least for this case!

File: docs/faq.rst (hunk 2 above)
Comment (jakevdp) [reply docs/faq.rst]: Thanks - done. PTAL!

**Review by skye — APPROVED** [review]

**Pull request merged**